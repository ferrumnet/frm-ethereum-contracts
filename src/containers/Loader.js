import React from "react";
import { connect } from "react-redux";
import IconLoader from "../components/common/Loader";

export const Loader = ({ loading }) =>
  loading ? (
    <div className="ui-loader">
      <IconLoader />
    </div>
  ) : null;

const mapStateToProps = ({ contract }) => ({
  loading: contract.loading
});

export default connect(
  mapStateToProps,
  null
)(Loader);
