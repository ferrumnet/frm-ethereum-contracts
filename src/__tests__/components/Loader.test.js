import Loader from "../../components/common/Loader";
import { shallow } from "enzyme";
import React from "react";

describe("Test Loader Component", () => {
  test("It should match the snapshot", () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
