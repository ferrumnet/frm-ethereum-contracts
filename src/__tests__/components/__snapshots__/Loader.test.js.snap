// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loader Component It should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IconLoader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <g
        fill="none"
        fillRule="evenodd"
      >
        <g
          strokeWidth="2"
          transform="translate(1 1)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            strokeOpacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
          >
            <animateTransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </g>,
      "height": "38",
      "stroke": "#fff",
      "viewBox": "0 0 38 38",
      "width": "38",
      "xmlns": "http://www.w3.org/2000/svg",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <g
          strokeWidth="2"
          transform="translate(1 1)"
        >
          <circle
            cx="18"
            cy="18"
            r="18"
            strokeOpacity=".5"
          />
          <path
            d="M36 18c0-9.94-8.06-18-18-18"
          >
            <animateTransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>,
        "fill": "none",
        "fillRule": "evenodd",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />,
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>,
          ],
          "strokeWidth": "2",
          "transform": "translate(1 1)",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "cx": "18",
              "cy": "18",
              "r": "18",
              "strokeOpacity": ".5",
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />,
              "d": "M36 18c0-9.94-8.06-18-18-18",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "attributeName": "transform",
                "dur": "1s",
                "from": "0 18 18",
                "repeatCount": "indefinite",
                "to": "360 18 18",
                "type": "rotate",
              },
              "ref": null,
              "rendered": null,
              "type": "animateTransform",
            },
            "type": "path",
          },
        ],
        "type": "g",
      },
      "type": "g",
    },
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <g
          fill="none"
          fillRule="evenodd"
        >
          <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>
        </g>,
        "height": "38",
        "stroke": "#fff",
        "viewBox": "0 0 38 38",
        "width": "38",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <g
            strokeWidth="2"
            transform="translate(1 1)"
          >
            <circle
              cx="18"
              cy="18"
              r="18"
              strokeOpacity=".5"
            />
            <path
              d="M36 18c0-9.94-8.06-18-18-18"
            >
              <animateTransform
                attributeName="transform"
                dur="1s"
                from="0 18 18"
                repeatCount="indefinite"
                to="360 18 18"
                type="rotate"
              />
            </path>
          </g>,
          "fill": "none",
          "fillRule": "evenodd",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <circle
                cx="18"
                cy="18"
                r="18"
                strokeOpacity=".5"
              />,
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
              >
                <animateTransform
                  attributeName="transform"
                  dur="1s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>,
            ],
            "strokeWidth": "2",
            "transform": "translate(1 1)",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": "18",
                "cy": "18",
                "r": "18",
                "strokeOpacity": ".5",
              },
              "ref": null,
              "rendered": null,
              "type": "circle",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <animateTransform
                  attributeName="transform"
                  dur="1s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />,
                "d": "M36 18c0-9.94-8.06-18-18-18",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "attributeName": "transform",
                  "dur": "1s",
                  "from": "0 18 18",
                  "repeatCount": "indefinite",
                  "to": "360 18 18",
                  "type": "rotate",
                },
                "ref": null,
                "rendered": null,
                "type": "animateTransform",
              },
              "type": "path",
            },
          ],
          "type": "g",
        },
        "type": "g",
      },
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
